### DISCUSSION TITLE
How to filter a user's repositories by language?

### DISCUSSION THREAD

**Discussion by sajan45** [community/community#24003] [New to GitHub]
<p>Hi.</p>
<p>Since last few months, I don’t see the language filter options for any user’s repo. Please refer to the attached screenshot below. It shows only the ‘All’ option. When I am not logged in to Github, I can see all of the languages to filter by in this dropdown but options disappear after I log in. I checked in multiple systems, browser too. I don’t remember changing any settings recently. Please help me in finding what is causing this.</p>
<p>Thank you.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/300e3cf3163a6b50a4da037887bd94ca4755f6ef.png" data-download-href="/uploads/short-url/6R7qg4yvrteHagmylfmVEDAWF0X.png?dl=1" title="Screenshot 2020-04-07 at 12.27.28 PM.png" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10629412/181080575-cec118a7-f709-4b1d-8b7c-072fc41ffc69.png" alt="Screenshot 2020-04-07 at 12.27.28 PM.png" data-base62-sha1="6R7qg4yvrteHagmylfmVEDAWF0X" width="690" height="271" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/300e3cf3163a6b50a4da037887bd94ca4755f6ef_2_690x271.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/300e3cf3163a6b50a4da037887bd94ca4755f6ef_2_1035x406.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/300e3cf3163a6b50a4da037887bd94ca4755f6ef.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/300e3cf3163a6b50a4da037887bd94ca4755f6ef_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-04-07 at 12.27.28 PM.png</span><span class="informations">1236×486 19.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/sajan45">@sajan45</a> - Thanks for reporting this! The <code>Language</code> dropdown should be present when viewing a user profile’s <code>Repositories</code> tab, whether you’re logged in or not.</p>
<p>I checked on our end and there’s not been any reports like this in the last month. If you’re still experiencing this behavior, could you share a screenshot of what you’re seeing when you’re logged in and viewing a user or organization account profile’s <code>Repositories</code>? Also, sharing the specific URL would help us investigate further!</p>

**Comment by sajan45** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> - Thank you for looking into this, my apologies for the late response. I still can’t any other options in the Language dropdown, not even when I am not logged in. Previously I think I was able to see options when not logged in. I checked all pages again, previously I did not check my own profile page and organization’s page. Language dropdown <strong>shows </strong> all options for my own profile and organization’s page, but it does not show options for any other user irrespective of the fact whether I follow them or not. I am attaching screenshots I took today. You can see that only 3 screenshots have options in language dropdown, i.e 2 organization’s page, and my own profile page. I tried in both Firefox and Chrome, in incognito mode too, to be sure no other thing like extensions is causing this. And one last thing, I can’t see language options of my own profile <a href="https://github.com/sajan45" rel="nofollow noopener">https://github.com/sajan45</a> too when  <strong>not logged in</strong> , I can see options in my own profile only when I am logged In.</p>
<p>Thank you.</p>
<p>Regards,</p>
<p>Sajan</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/03a4c61a8687ae0f6901a969c582cc72335227ca.png" data-download-href="/uploads/short-url/wesroDEr18rurow7pDXya32fns.png?dl=1" title="user-ruslan.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080577-a7522e9a-90c6-4351-b159-a7a0c36cbfd9.png" alt="user-ruslan.png" data-base62-sha1="wesroDEr18rurow7pDXya32fns" width="690" height="392" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/03a4c61a8687ae0f6901a969c582cc72335227ca_2_690x392.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/03a4c61a8687ae0f6901a969c582cc72335227ca_2_1035x588.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/03a4c61a8687ae0f6901a969c582cc72335227ca_2_1380x784.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/03a4c61a8687ae0f6901a969c582cc72335227ca_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">user-ruslan.png</span><span class="informations">1431×813 200 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/01426e2361076cd01147a0e62bf82059c7f98fa2.png" data-download-href="/uploads/short-url/b8NOWGQcRysSwOagbYUlgQtqaS.png?dl=1" title="user-chris.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080579-348e25a8-3060-4bb3-8836-dc9dd696d329.png" alt="user-chris.png" data-base62-sha1="b8NOWGQcRysSwOagbYUlgQtqaS" width="690" height="381" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/01426e2361076cd01147a0e62bf82059c7f98fa2_2_690x381.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/01426e2361076cd01147a0e62bf82059c7f98fa2_2_1035x571.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/01426e2361076cd01147a0e62bf82059c7f98fa2_2_1380x762.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/01426e2361076cd01147a0e62bf82059c7f98fa2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">user-chris.png</span><span class="informations">1424×787 204 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/de7b3ecba2fc290bfd465fa30503b6168c7faa1f.jpeg" data-download-href="/uploads/short-url/vKa37dLKTUlgDG7AWhnMOp4RZpB.jpg?dl=1" title="user-andrew.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080580-755a911f-d55d-4cc8-bcee-c64009bac8e9.jpg" alt="user-andrew.png" data-base62-sha1="vKa37dLKTUlgDG7AWhnMOp4RZpB" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/de7b3ecba2fc290bfd465fa30503b6168c7faa1f_2_690x388.jpg, https://aws1.discourse-cdn.com/github/optimized/2X/d/de7b3ecba2fc290bfd465fa30503b6168c7faa1f_2_1035x582.jpg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/de7b3ecba2fc290bfd465fa30503b6168c7faa1f_2_1380x776.jpg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/de7b3ecba2fc290bfd465fa30503b6168c7faa1f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">user-andrew.png</span><span class="informations">1429×804 212 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e2d326d0d564c53c002147c0f48dbf87710be96e.png" data-download-href="/uploads/short-url/wmAivMglhuDDwvjLQKRhvfaUTX8.png?dl=1" title="org-rails.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080581-677f4f98-71ef-46e9-84d4-01700891d233.png" alt="org-rails.png" data-base62-sha1="wmAivMglhuDDwvjLQKRhvfaUTX8" width="690" height="405" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e2d326d0d564c53c002147c0f48dbf87710be96e_2_690x405.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e2d326d0d564c53c002147c0f48dbf87710be96e_2_1035x607.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e2d326d0d564c53c002147c0f48dbf87710be96e_2_1380x810.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e2d326d0d564c53c002147c0f48dbf87710be96e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">org-rails.png</span><span class="informations">1395×819 231 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a0bffa8d3be6ab06e903cb5263bf378159dbb4b3.png" data-download-href="/uploads/short-url/mW3HSH8I6ks1MDVEs9SW26gTQk3.png?dl=1" title="org-phoenix.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080583-64121087-95ac-4f51-9312-b0027f0676f1.png" alt="org-phoenix.png" data-base62-sha1="mW3HSH8I6ks1MDVEs9SW26gTQk3" width="690" height="381" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a0bffa8d3be6ab06e903cb5263bf378159dbb4b3_2_690x381.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a0bffa8d3be6ab06e903cb5263bf378159dbb4b3_2_1035x571.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/a/a0bffa8d3be6ab06e903cb5263bf378159dbb4b3_2_1380x762.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a0bffa8d3be6ab06e903cb5263bf378159dbb4b3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">org-phoenix.png</span><span class="informations">1419×785 159 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c9638f766105c17a69593152d2adea24b74bb167.png" data-download-href="/uploads/short-url/sJzhhs221RTbQO3le6weNLTMROD.png?dl=1" title="my-profile.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10629412/181080585-2544571a-fd32-4d01-8a29-508cdb932916.png" alt="my-profile.png" data-base62-sha1="sJzhhs221RTbQO3le6weNLTMROD" width="690" height="384" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c9638f766105c17a69593152d2adea24b74bb167_2_690x384.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c9638f766105c17a69593152d2adea24b74bb167_2_1035x576.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/c9638f766105c17a69593152d2adea24b74bb167_2_1380x768.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c9638f766105c17a69593152d2adea24b74bb167_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">my-profile.png</span><span class="informations">1432×798 126 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by requiredregistration** [comment]
<p>this is still broken. in the language list only All is available. whether one is logged in or not makes no difference.</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/sajan45">@sajan45</a> &amp; <a class="mention" href="/requiredregistration">@requiredregistration</a> - Thanks for following up! I checked this on both user profiles and organization profiles and can confirm that not all languages are listed.</p>
<p>On my end, I will report this to our engineering team. However, I can’t promise a timeline for when it will be reviewed, prioritized, and addressed.</p>
<p>In the meantime, it may be worth checking out our GraphQL API and using our <a href="https://docs.github.com/en/graphql/overview/explorer">GraphQL Explorer</a> to run a query that fetches a collection of a user’s repositories’ languages:</p>
<pre><code class="lang-auto">{
  user(login: "francisfuzz") {
    login
    repositories(first: 100, orderBy: {field: PUSHED_AT, direction: DESC}) {
      nodes {
        primaryLanguage {
          name
        }
      }
    }
  }
}
</code></pre>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/sajan45">@sajan45</a> &amp; <a class="mention" href="/requiredregistration">@requiredregistration</a> Good news! our team just shipped an update that ensures all languages are shown for all user profiles. Could I confirm with both of you that you’re no longer seeing the previously reported behavior?</p>
<p>For example, <a href="https://github.com/substack?tab=repositories"><code>substack</code>'s repositories tab</a>’s language dropdown shows all recognized languages for their repository:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/5/e5a6c9015f69ca70ef51aa09551d3b986772d916.gif" data-download-href="/uploads/short-url/wLAFVNEqhXkARMTuC5roRQNhLp4.gif?dl=1" title="substack-repos-tab-language-dropdown"><img src="https://user-images.githubusercontent.com/15894826/181080590-1690e19c-779c-401f-bc27-7de5a3bede15.gif" alt="substack-repos-tab-language-dropdown" data-base62-sha1="wLAFVNEqhXkARMTuC5roRQNhLp4" width="690" height="382" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/5/e5a6c9015f69ca70ef51aa09551d3b986772d916_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">substack-repos-tab-language-dropdown</span><span class="informations">1914×1060 2.61 MB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by requiredregistration** [comment]
<p>thank you <a class="mention" href="/francisfuzz">@francisfuzz</a>. i just checked and it has been fixed.</p>

**Comment by sajan45** [comment, marked as answer]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> It has been fixed. Thank you.</p>
